btree/btree.c

```diff
@@ -3468,14 +3471,21 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 			 */
 			if (!node_isvalid(lev->l_node) ||
 			    !node_verify(lev->l_node)) {
-				m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_SETUP);
+				return m0_sm_op_sub(&bop->bo_op, P_CLEANUP,
+						    P_SETUP);
 			}
 			oi->i_nop.no_node = NULL;
 
 			oi->i_key_found = node_find(&node_slot,
 						    &bop->bo_rec.r_key);
 			lev->l_idx = node_slot.s_idx;
 			if (node_level(node_slot.s_node) > 0) {
+				if (oi->i_used == oi->i_height - 1) {
+					/* If height of tree increased. */
+					return m0_sm_op_sub(&bop->bo_op,
+							    P_CLEANUP, P_SETUP);
+				}
```

Comment:
Better to relocate this check below right after the oi->i_used++ on line 3498. With this the code become easier to understand and guess. Also after the increment of i_used the check will be if (oi->i_used >= ot->i_height)

Reply:
Then we unnecessarily have to perform node_child() and address_in_Segment() check . To avoid this if height increased, I am doing check at line 3483

Reply:
That is fine but we need to make sure the code looks nice and clean and compartmentalized.
Also again this is not a very frequent occurrence where the height would have changed so please make the change as I suggested as long as this move is not going to cause any crashes.